client/lib/purchases/reducer.js

```diff
@@ -1,165 +0,0 @@
-/** @format */
-
-/**
- * External dependencies
- */
-
-import { assign, find, matches } from 'lodash';
-
-/**
- * Internal Dependencies
- */
-import { action as ActionTypes } from 'lib/upgrades/constants';
-
-/**
- * Constants
- */
-const initialState = {
-	data: [],
-	error: null,
-	isFetchingSitePurchases: false,
-	isFetchingUserPurchases: false,
-	hasLoadedSitePurchasesFromServer: false,
-	hasLoadedUserPurchasesFromServer: false,
-};
-
-function updatePurchaseById( state, id, properties ) {
-	return assign( {}, state, {
-		data: state.data.map( purchase => {
-			if ( id === purchase.id ) {
-				return assign( {}, purchase, properties );
-			}
-			return purchase;
-		} ),
-	} );
-}
-
-/**
- * Overwrites the purchases in the store with the purchases from the new purchases array
- * that share the same `id` value.
- *
- * @param {array} existingPurchases - an array of purchases in the store
- * @param {array} newPurchases - an array of purchases fetched from the API
- * @return {array} An array of purchases
- */
-function overwriteExistingPurchases( existingPurchases, newPurchases ) {
-	let purchases = newPurchases;
-
-	existingPurchases.forEach( purchase => {
-		if ( ! find( purchases, { id: purchase.id } ) ) {
-			purchases = purchases.concat( purchase );
-		}
-	} );
-
-	return purchases;
-}
-
-/**
- * Removes purchases that are missing from the new purchases array and match the given predicate.
- *
- * @param {array} existingPurchases - an array of purchases in the store
- * @param {array} newPurchases - an array of purchases fetched from the API
- * @param {object} predicate - the predicate to check before removing the item from the array.
- * @return {array} An array of purchases
- */
-function removeMissingPurchasesByPredicate( existingPurchases, newPurchases, predicate ) {
-	return existingPurchases.filter( purchase => {
-		if ( matches( predicate )( purchase ) && find( newPurchases, { id: purchase.id } ) ) {
-			// this purchase is present in the new array
-			return true;
-		}
-
-		if ( ! matches( predicate )( purchase ) ) {
-			// only overwrite remove purchases that match the predicate
-			return true;
-		}
-
-		// the purchase doesn't match the predicate or is missing from the array of new purchases
-		return false;
-	} );
-}
-
-function updatePurchases( existingPurchases, action ) {
-	let purchases, predicate;
-
-	if ( ActionTypes.PURCHASES_SITE_FETCH_COMPLETED === action.type ) {
```

Comment:
Seems like many of those legacy actions (or probably all of them that start with `PURCHASE`) are unused but still defined in https://github.com/Automattic/wp-calypso/blob/master/client/lib/upgrades/constants.js. Perhaps we could also clean those up?

Reply:
I took a look and it seems like all of the following (removed in this PR) are used elsewhere:

```
PRIVACY_PROTECTION_CANCEL_COMPLETED
PRIVACY_PROTECTION_CANCEL_FAILED
PURCHASE_REMOVE_COMPLETED
PURCHASE_REMOVE_FAILED
PURCHASES_REMOVE
PURCHASES_SITE_FETCH
PURCHASES_SITE_FETCH_COMPLETED
PURCHASES_SITE_FETCH_FAILED
PURCHASES_USER_FETCH
PURCHASES_USER_FETCH_COMPLETED
PURCHASES_USER_FETCH_FAILED
```

Reply:
Most of these are actually not used - the confusion probably comes that for some of them we have the same names, defined as Redux action types in `client/state/action-types.js`.

Reply:
🤦‍♂️ You're probably right… this is nasty duplication. I'll try to track some down.